### DISCUSSION TITLE
How can I monitor the potential of anode surface?

### DISCUSSION THREAD

**Discussion by gao-yidan** [pybamm-team/PyBaMM#2360] [Q&A]
Hi everyone!

I am a newbee to this Pybamm model. I am now trying to monitor the potential at the interphase between anode and separator, in order to model the lithium plating. Could someone tell me how to plot this potential comparing to Li/Li+. I couldn't find it in the output variables. 

Thank you in advance! I really appreciate your time!

**Comment by Sunnyturicum** [comment]
I have a similar issue, which is why I post my question in here: 

### Goal
I am trying to plot the anode potential similar to the following picture.


![image](https://github.com/pybamm-team/PyBaMM/assets/63863005/360f41e0-7b45-415a-b76f-7fa4eaba87a8)

### Minimal working example
```
model = pybamm.lithium_ion.SPMe(
    {
        "lithium plating": "partially reversible",
        "lithium plating porosity change": "true",  # alias for "SEI porosity change"
    }
)

pybamm.set_logging_level("NOTICE")
cycle_number = 50
exp = pybamm.Experiment(
    [("Discharge at 1C until 2.5 V",  # ageing cycles
    "Charge at 0.3C until 3.65 V",
    "Hold at 3.65 V until C/20",
    "Rest for 4 hours",)] * cycle_number
)

simulations = []
temperatures = [268.15, 288.15] #-5, 15°C

solutions = {} # dict, key is temp and values is solution object

for temp in temperatures:
    param = prada2013_params
    param.update({"Ambient temperature [K]": temp})

    sim = pybamm.Simulation(model, parameter_values=param, experiment=exp)
    safe_solver = pybamm.CasadiSolver(mode="safe", dt_max=120)
    sol = sim.solve(solver=safe_solver, calc_esoh=False)
    solutions[temp] = sol

x_axis_var_str = "Time [h]"
y_axis_var_str = "Negative electrode potential [V]"

first_cycle = 0
last_cycle = cycle_number-1

# create figure
fig, ax = plt.subplots(figsize=(8, 6))


first_cycle_colors = ["steelblue","lightblue", "coral", "red"]  # Choose colors for first cycles
last_cycle_colors = ["steelblue","lightblue", "coral", "yellow"]  # Choose colors for last cycles



for i, (temp, sol) in enumerate(solutions.items()):
    for j, cycle in enumerate([first_cycle, last_cycle]):
        # Plot over distances at time t
        #plt.plot(sol.cycles[cycle][y_axis_var_str].data[:, 50])
        
        time = sol.cycles[cycle][x_axis_var_str].data
        time -= time.min()
        potential = sol.cycles[cycle][y_axis_var_str].data[-1] # -1 for furthest point
        
        
        if cycle == first_cycle:
            line_style = "-"
            label = "{} °C - cycle {}".format(temp-273.15, cycle + 1)
            color = first_cycle_colors[i]
        else:
            line_style = "--"
            label = "{} °C - cycle {}".format(temp -273.15, cycle + 1)
            color = last_cycle_colors[i]

        ax.plot(time, potential, color=color, label=label, linestyle=line_style)


ax.set_xlabel(x_axis_var_str)
ax.set_ylabel(y_axis_var_str)
ax.legend()
plt.tight_layout()
plt.show()


```

For my parameter set, I mixed prada2013 with degradation parameters from oKane2022. Using 
```
prada2013_params = pybamm.ParameterValues("OKane2022")
```
should work.

### Problem

In my understanding, the time during which lithium plating occurs increases as the cycling proceeds. This means the negative electrode potential (at the anode/separator interface) should over time have an increasing amount below 0 V. However, from my graph, I can't come to this conclusion. Does somebody has any advice?


![image](https://github.com/pybamm-team/PyBaMM/assets/63863005/05a1bafe-7f24-4cae-9bef-6ab3200356ce)

**Reply by brosaplanella** [reply]
I think that you want the `"Negative electrode surface potential difference [V]"` rather than `"Negative electrode potential [V]"`, as the former is the one that dictates how strong is the plating reaction.

**Reply by Sunnyturicum** [reply]
Thank you for this input. I adapted the variable accordingly. However, the potential always stays above 0 V. Am I still doing something wrong? Thanks in advance for your support. Always much appreciated.

![image](https://github.com/pybamm-team/PyBaMM/assets/63863005/c754d596-86ec-4063-b51b-60f6d4f32ef5)

**Reply by brosaplanella** [reply]
That could be a parameter issue (i.e. your cell is aging quite slowly). Try making the exchange current density for plating larger...

**Reply by Sunnyturicum** [reply]
Thank you, I adapted the electrode width. The potential is now negative. You were right that this is probably due to a parameter issue.

**Comment by Dorothy-Reyes** [comment]
Hi,
I had the same issue when working with the Pybamm model and trying to monitor the potential at the anode-separator interface for lithium plating. To solve this problem, you can access the potential difference between the anode and Li/Li+ in Pybamm. Here's how you can do it:

Define a new output variable in your Pybamm model, let's call it "Anode Potential vs. Li/Li+".

Use the expression: "anode potential - Li/Li+ potential" to calculate the potential difference at the anode-separator interface.

Add this new output variable to your simulations and plots.

This way, you'll be able to monitor and plot the potential at the anode-separator interface relative to Li/Li+. Good luck with your modelling!

**Reply by Sunnyturicum** [reply]
Thanks a lot for your support!
